changelog.d/13573.misc

```diff
@@ -0,0 +1 @@
+Cache user IDs instead of profiles to reduce cache memory usage (renames cache `_get_joined_profile_from_event_id` to `_get_user_id_from_membership_event_id`). Contributed by Nick @ Beeper (@fizzadar).
```

Comment:
I'm really cautious about renaming the caches if I'm honest, especially for something like this. I wonder if we should explicitly keep the old name, by adding a `name` param to `@cached`?

Reply:
Yeah totally makes sense, esp. if anyone has tweaked the cache factor (like us). Added in https://github.com/matrix-org/synapse/pull/13573/commits/e49e71634c59ddcd38d8d876b7348bdb41177879 & https://github.com/matrix-org/synapse/pull/13573/commits/0c54e9a9d6bf3166ebd4ad9559eafeaee4b0372c.

Reply:
Speedy!